README.md

```diff
@@ -200,17 +200,40 @@ Properties passed to the `w` function represent the public API for a widget.
 
 The properties lifecycle starts when properties are passed to the widget.
 The properties lifecycle is performed in the widgets `setProperties` function.
-This function uses the widget instance's `diffProperties` function to determine whether any of the properties have changed since the last render cycle.
-By default `diffProperties` provides a shallow comparison of the previous properties and new properties.
 
-The `diffProperties` function is also responsible for creating a copy (the default implementation uses`Object.assign({}, newProperties)` of all changed properties.
-The depth of the returned diff is equal to the depth used during the equality comparison.
+This function uses the widget's `diffProperty` and `diffProperties` decorators to determine whether any of the properites have changed since the last render cycle. **By default, if `@diffProperty` or `@diffProperties` are not used, changing a property will not cause the widget to re-render.**
 
-<!-- add example of 'depth' -->
+##### Catch-all property diff control
 
-**Note:** If a widget's properties contain complex data structures that you need to diff, then the `diffProperties` function will need to be overridden.
+The `@diffProperties` decorator will provide a generic diff implementation for _all_ the properties on a widget. By default `diffProperties` uses `DiffType.AUTO` to compare properties:
 
-##### Custom property diff control
+```typescript
+@diffProperties // <-- uses DiffType.AUTO
+class MyWidget extends WidgetBase<WidgetProperties> {
+}
+```
+
+But this can be changed to any `DiffType`:
```

Comment:
Yes! I assumed it was documented already, but that might have been a poor assumption 😄